app/screens/gallery/renderers/video/video_controls/progress_bar.tsx

```diff
@@ -0,0 +1,181 @@
+import Animated, {
+    useAnimatedStyle,
+    useSharedValue,
+    runOnJS,
+    withTiming,
+    interpolate,
+    runOnUI,
+} from 'react-native-reanimated';
+
+interface ProgressBarProps {
+    progress: number;
+    duration: number;
+    onSeek: (time: number) => void;
+}
+
+const styles = StyleSheet.create({
+    container: {
+        top: 4,
+        marginBottom: 8,
+        paddingHorizontal: 4,
+        width: '100%',
+        flex: 1,
+    },
+    touchArea: {
+        height: 24,
+        justifyContent: 'center',
+        position: 'relative',
+    },
+    track: {
+        backgroundColor: 'rgba(255, 255, 255, 0.3)',
+        position: 'absolute',
+        left: 0,
+        right: 0,
+    },
+    bar: {
+        backgroundColor: 'white',
+        position: 'absolute',
+        left: 0,
+    },
+    thumb: {
+        position: 'absolute',
+        backgroundColor: 'white',
+        top: '50%',
+        marginTop: -8,
+        shadowColor: '#000',
+        shadowOffset: {width: 0, height: 2},
+        shadowOpacity: 0.3,
+        shadowRadius: 4,
+        elevation: 4,
+    },
+});
+
+const ProgressBar: React.FC<ProgressBarProps> = ({
+    progress,
+    duration,
+    onSeek,
+}) => {
+    const progressRef = useRef<View>(null);
+    const progressWidth = useSharedValue(0);
+    const dragPosition = useSharedValue(0);
+    const trackHeight = useSharedValue(4);
+    const thumbOpacity = useSharedValue(0);
+    const isDraggingShared = useSharedValue(false);
+    const localProgress = useSharedValue(progress);
+
+    React.useEffect(() => {
+        runOnUI(() => {
+            'worklet';
+            if (!isDraggingShared.value) {
+                localProgress.value = progress;
+            }
+        })();
+
+    // no need to add shared values to the dependency array
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [progress]);
+
+    const startDragging = () => {
+        'worklet';
+        isDraggingShared.value = true;
+        trackHeight.value = withTiming(8, {duration: 150}); // Double height
+        thumbOpacity.value = withTiming(1, {duration: 150});
+    };
+
+    const stopDragging = () => {
+        isDraggingShared.value = false;
+        trackHeight.value = withTiming(4, {duration: 150}); // Back to normal
+        thumbOpacity.value = withTiming(0, {duration: 150});
+        localProgress.value = progress;
+    };
+
+    const progressGesture = Gesture.Pan().
+        minDistance(0).
+        activeOffsetX([-5, 5]).
+        onStart((event) => {
+            startDragging();
+            dragPosition.value = event.x;
+        }).
+        onUpdate((event) => {
+            dragPosition.value = Math.max(0, Math.min(progressWidth.value, event.x));
+            const newProgress = dragPosition.value / progressWidth.value;
+            localProgress.value = newProgress;
+            const newTime = newProgress * duration;
+            runOnJS(onSeek)(newTime);
```

Comment:
I started looking into video scrubbing performance a bit (by no means an expert, so bear with me). But iiuc, is this is called every frame?
If it is, maybe we should throttle it to something like 10 or 15 times a second?

Reply:
ahhhh you are absolutely spot on, I'm going to throttle every 100ms